src/Microsoft.Owin.Security.Authorization.Mvc/ResourceAuthorizeAttribute.cs

```diff
@@ -29,9 +29,9 @@ public override void OnAuthorization(System.Web.Mvc.AuthorizationContext filterC
                 throw new ArgumentNullException(nameof(filterContext));
             }
 
-            if (!filterContext.HttpContext.Items.Contains(s_controllerKey))
+            if (!filterContext.HttpContext.Items.Contains(s_authorizationContextKey))
             {
-                filterContext.HttpContext.Items.Add(s_controllerKey, filterContext.Controller);
+                filterContext.HttpContext.Items.Add(s_authorizationContextKey, filterContext.Controller);
```

Comment:
This should pass the entire filterContext, not filterContext.Controller;
```suggestion
                filterContext.HttpContext.Items.Add(s_authorizationContextKey, filterContext);
```